src/server/bg_services/namespace_monitor.js

```diff
@@ -214,13 +218,15 @@ class NamespaceMonitor {
             await nb_native().fs.readdir(fs_context, nsr.nsfs_config.fs_root_path);
             const stat = await nb_native().fs.stat(fs_context, nsr.nsfs_config.fs_root_path);
             //In the event of deleting the nsr.nsfs_config.fs_root_path in the FS side, 
-            // The number of link will be 0, then we will throw ENOENT which translate to STORAGE_NOT_EXIST
+            // The number of link will be 0, then we will throw an error which translate to STORAGE_NOT_EXIST
             if (stat.nlink === 0) {
-                throw Object.assign(new Error('FS root path has no links'), { code: 'ENOENT' });
+                throw Object.assign(new Error('FS root path has no links'), { code: internal_err_code });
             }
         } catch (err) {
             dbg.log1('test_nsfs_resource: got error:', err, nsr.nsfs_config);
-            throw err;
+            dbg.log1(`test_nsfs_resource: we change the code to control the mapping in pool server` +
```

Comment:
I think that we need to refactor this whole error handling of this function - 
1. nlink === 0 - should still throw ENOENT 
2. inside the catch - 
    2.1. if err.code === ENOENT (it can happen because of nlink ===0 or stat/readdir threw ENOENT) we should throw MissingRootPathError.
    2.2. if err.code === EPERM || EACCESS we should throw AccessDenied
    else - throw err

Reply:
we can throw an internal error but we need to think what are the consequences and to which phase we will map it and when.. not on this PR anyway

Reply:
I'll mention this comment as a gap, resolving.